Lib/test/test_queue.py

```diff
@@ -578,13 +577,19 @@ def _shutdown_put_join(self, immediate):
         for func, params in thrds:
             threads.append(threading.Thread(target=func, args=params))
             threads[-1].start()
-        self.assertEqual(q.unfinished_tasks, nb)
-        for i in range(nb):
-            t = threading.Thread(target=q.task_done)
-            t.start()
-            threads.append(t)
+        self.assertEqual(q.unfinished_tasks, 1)
+
         q.shutdown(immediate)
         go.set()
+
+        if immediate:
+            with self.assertRaises(self.queue.ShutDown):
+                q.get_nowait()
+        else:
+            task = q.get()
+            self.assertEqual(task, "Y")
```

Comment:
I wouldn't name this variable task -- rather, "result", since elsewhere the values retrieved from the queue are stored in a list named "results".
```suggestion
            result = q.get()
            self.assertEqual(result, "Y")
```